pep-0567.rst

```diff
@@ -133,21 +133,36 @@ the context variable in the current ``Context``::
     # Set the variable 'var' to 1 in the current context.
     var.set(1)
 
-``contextvars.Token`` is an opaque object that should be used to
-restore the ``ContextVar`` to its previous value, or remove it from
-the context if it was not set before.  The ``ContextVar.reset(Token)``
-is used for that::
+``ContextVar.reset(token)`` is used to reset the variable in the
+current context to the value it had before the ``set()`` operation
+that created the ``token``::
+
+    assert var.get(None) is None
 
     token = var.set(1)
     try:
         ...
     finally:
         var.reset(token)
 
+    assert var.get(None) is None
+
 ``ContextVar.reset()`` method is idempotent and can be called
 multiple times on the same Token object: second and later calls
 will be no-ops.
 
+
+contextvars.Token
+-----------------
+
+``contextvars.Token`` is an opaque object that should be used to
+restore the ``ContextVar`` to its previous value, or remove it from
+the context if the variable was not set before.  It can be created
+only by calling ``ContextVar.set()``.
+
+For debug and introspection purposes it has a read-only attribute
+``Token.var`` pointing to the variable that created the token.
```

Comment:
Wouldn't it be useful if it also published the old value? That, too would be helpful for debugging. And if there is no old value it should publish that fact too.

Reply:
The best way would probably be to have:

* `Token.MISSING` as a special marker object;

* `Token.value` that points to either a value that was in the context before `set()`, or to `Token.MISSING`.

Would that work?